app/src/main/java/com/babylon/wallet/android/domain/usecases/p2plink/EstablishP2PLinkConnectionUseCase.kt

```diff
@@ -70,18 +69,25 @@ class EstablishP2PLinkConnectionUseCase @Inject constructor(
                 override suspend fun completeLinking(connectionId: String): Result<Unit> {
                     val p2pLinks = p2PLinksRepository.getP2PLinks()
 
-                    val walletClientKeyBytes = p2pLinks.findBy(payload.publicKey.hex)?.walletPrivateKey
-                        ?.decodeHex() ?: PrivateKey.EddsaEd25519.newRandom().toByteArray()
+                    val walletClientKeyBytes = encryptedPreferencesManager.getP2PLinksWalletPrivateKey()
+                        ?.decodeHex()
+                        ?: run {
+                            val newPrivateKey = PrivateKey.EddsaEd25519.newRandom().toByteArray()
```

Comment:
Hmmm PrivateKey has been deleted in sargon integration. An example of that was that we replaced the ephemeral private key [we created in transaction review](https://github.com/radixdlt/babylon-wallet-android/pull/895/files#diff-15f75dfc84cca5713090177185b79a496533dccd41fae3fee803cbcb0acd0f52R83) with a custom sargon object called `NotaryPrivateKey` which has a `secureRandom()` factory. 

Maybe this needs to be done in here too.

Reply:
This is how NotaryPrivateKey it is implemented in sargon https://github.com/radixdlt/sargon/blob/develop/jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/NotaryPrivateKey.kt